cmd/infra/ibmcloud_powervs/create.go

```diff
@@ -627,50 +632,36 @@ func (infra *Infra) setupPowerVsCloudConnection(option *CreateInfraOptions, sess
 	return nil
 }
 
-func (infra *Infra) monitorCloudConnection(option *CreateInfraOptions, session *ibmpisession.IBMPISession) (err error) {
+func (infra *Infra) monitorCloudConnection(session *ibmpisession.IBMPISession) (err error) {
 	client := instance.NewIBMPICloudConnectionClient(context.Background(), session, infra.PowerVSCloudInstanceID)
 	var cloudConn *models.CloudConnection
-	retryExceeds := false
-	ttl := make(chan bool, 1)
-	go func() {
-		for {
-			if retryExceeds {
-				break
-			}
 
-			cloudConn, err = client.Get(infra.PowerVSCloudConnectionID)
-
-			if err != nil {
-				break
-			}
+	f := func() (bool, error) {
+		cloudConn, err = client.Get(infra.PowerVSCloudConnectionID)
+		if err != nil {
+			return false, err
+		}
 
-			if cloudConn != nil {
-				log.Log.Info("Waiting for Cloud Connection to up", "id", cloudConn.CloudConnectionID, "status", cloudConn.LinkStatus)
-				if *cloudConn.LinkStatus == "established" {
-					break
-				}
+		if cloudConn != nil {
+			log.Log.Info("Waiting for Cloud Connection to up", "id", cloudConn.CloudConnectionID, "status", cloudConn.LinkStatus)
+			if *cloudConn.LinkStatus == establishedState {
+				return true, nil
 			}
-			time.Sleep(5 * time.Second)
 		}
-		ttl <- true
-	}()
 
-	select {
-	case <-time.After(30 * time.Minute):
-		retryExceeds = true
-		return fmt.Errorf("time exceeds waiting for cloud connection to up")
-	case <-ttl:
+		return false, nil
 	}
 
+	err = wait.PollImmediate(time.Second*5, time.Minute*10, f)
```

Comment:
can we move move these values to constant for better readability.

Reply:
made them as const.